### PR TITLE
Mrtk upgrade guide

### LINKED ISSUE #882: 2017.2 Upgrade Guide
We need a document that describes all the breaking changes and best practices for upgrading a project from 2017.1 to 2017.2

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixed #882 Preview [Here](https://github.com/StephenHodgson/MixedRealityToolkit-Unity/blob/MRTK-UpgradeGuide/UpgradeGuide.md)

Also updated the roadmap.

### CODE REVIEW

**Comment by Zod-** [comment]
Should probably mention that the behavior of `SetGlobalListeners` has changed. It's now getting registered/unregistered with OnEnable/OnDisable.

**Comment by StephenHodgson** [comment]
@keveleigh did you see anything you'd want to add to the upgrade guide?

File: UpgradeGuide.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## 2017.1 Upgrade Guide
+
+### What you'll need
+
+- [Unity Editor 2017.1.2f1](https://unity3d.com/unity/whats-new/unity-2017.1.2) or later
+- Window Build 1603 or later
+- Visual Studio 2015 or later
+- Windows 10 SDKs 10.0.10240.0 though 10.0.15063.0
+
+### List of braking changes
```
Comment (keveleigh) [inline UpgradeGuide.md]: typo: breaking

**Comment by keveleigh** [comment]
@StephenHodgson I think that looks good! I can't think of anything else at the moment.

Would this be the right place for things like "If you're currently using the GazeManager, you might want to update to using FocusManager to add support for motion controllers as well." etc.

**Comment by StephenHodgson** [comment]
If people are using the InputManager prefab then they'll automatically start using the `FocusManager` without issue.

I added `Obsolete` warnings to update their usages to `FocusManager` in code.

**Comment by keveleigh** [comment]
I'm referring to usage of `GazeManager.Instance.[whatever]` in their own code, like [GazeableColorPicker](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit-Examples/ColorPicker/GazeableColorPicker.cs#L30). That should really be updated to use FocusManager. I actually think I have a branch where I started migrating this haha...

**Comment by StephenHodgson** [comment]
Yeah I fixed all the test scenes to use the updated usages.  Yeah All of the methods that moved from Gaze to Focus managers have `Obsolete` tags with the suggested new usage.

**Comment by keveleigh** [comment]
> I fixed all the test scenes to use the updated usages

In the repo? Or are you referring to incoming changes? The example I linked above is still using GazeManager in both branches.

**Comment by StephenHodgson** [comment]
The `GazeManager` is still a used component.

**Comment by keveleigh** [comment]
@StephenHodgson `GazeManager` as a directly referenced component should be obsolete. `FocusManager` should be used as the abstraction layer instead.

**Comment by StephenHodgson** [comment]
If that was the case, then the `GazeManager` would have been made private or been removed from the dev branch, correct?

**Comment by StephenHodgson** [comment]
We can still work our way to that.

**Comment by keveleigh** [comment]
Sounds good to me! It was made an `IPointingSource` to work with FocusManager. No longer making it a `Singleton` may be the next step, though knowing where a user is looking vs where they're pointing may be interesting.

I think working towards `FocusManager` being the main way to detect focused objects and whatnot is the right thing to do though, so it's easier to add more pointing sources in the future and apps become forward compatible, instead of just relying on gaze.

**Comment by StephenHodgson** [comment]
I'm surprised no one just renamed the class and made the changes right there.

**Comment by keveleigh** [comment]
Gaze is one of several possible pointers, so I'm not sure what you mean. We would still want an `IPointingSource` that gets its data from the camera, and that's what `GazeManager` was converted into.

**Comment by StephenHodgson** [comment]
Yeah you're right. Wouldn't have worked that way.

**Pull request merged**